source/mesh_deformation/fastscapecc.cc

```diff
@@ -0,0 +1,1061 @@
+            }
+
+          // Then also retrieve what the other processes sent to us and put
+          // them into the same arrays as well:
+          for (unsigned int p = 1; p < Utilities::MPI::n_mpi_processes(this->get_mpi_communicator()); ++p)
+            {
+// TODO: In order to send/receive the std::map or std::multimap mentioned above, you can use
+// Utilities::MPI::isend/irecv. These are easier to use than what you have here because
+// they work on *any* data structure.
+
+              MPI_Status status;
+              MPI_Probe(p, 42, this->get_mpi_communicator(), &status);
+              int incoming_size = 0;
+              MPI_Get_count(&status, MPI_DOUBLE, &incoming_size);
+
+              for (unsigned int i = 0; i < temporary_variables.size(); ++i)
+                {
+                  temporary_variables[i].resize(incoming_size);
+                }
+
+              for (unsigned int i = 0; i < temporary_variables.size(); ++i)
+                MPI_Recv(&temporary_variables[i][0], incoming_size, MPI_DOUBLE, p, 42, this->get_mpi_communicator(), &status);
+
+              for (unsigned int i = 0; i < temporary_variables[1].size(); ++i)
+                {
+                  int index = static_cast<int>(temporary_variables[1][i]);
+                  h[index] = temporary_variables[0][i];
+                  vz[index] = temporary_variables[2][i];
+                }
+            }
+
+          // Save the current elevation so that we can later take the difference to
+          // the one after calling FastScape:
+          const std::vector<double> h_old = h;
+
+          // std::cout<<"here it works 11"<<std::endl;
+
+          const double aspect_timestep_in_years = this->get_timestep() / year_in_seconds;
+
+          unsigned int fastscape_iterations = fastscape_steps_per_aspect_step;
+          double fastscape_timestep_in_years = aspect_timestep_in_years / fastscape_iterations;
+          while (fastscape_timestep_in_years > maximum_fastscape_timestep)
+            {
+              fastscape_iterations *= 2;
+              fastscape_timestep_in_years *= 0.5;
+            }
+
+          std::cout << "[DEBUG] fastscape_steps_per_aspect_step = " << fastscape_steps_per_aspect_step << std::endl;
+          std::cout << "[DEBUG] aspect_timestep_in_years = " << aspect_timestep_in_years << std::endl;
+          std::cout << "[DEBUG] fastscape_iterations = " << fastscape_iterations << std::endl;
+            
+
+// TODO: This is a pretty arbitrary cell, with a pretty arbitrary cell diameter. Is that what you
+// want? If so, perhaps say so in a comment and explain what it is used for.
+//
+// Separately, let's call this variable surface_cell_area in concordance with our convention
+          const double surface_cell_area = surface_mesh.begin_active()->measure();
+
+          // 1. Create the FastScape grid adapter
+// TODO: I haven't looked at the GridAdapter yet, but we should see whether we can't get rid of the const_cast.
```

Comment:
I think it should work without the `const_cast`? The Fastscapelib grid adapter class internally keeps a non-const reference to the underlying mesh (although the adapter class shouldn't change any state of the underlying mesh).